### PR TITLE
Add filter and localStorage

### PR DESCRIPTION
**Pull request by Baozzz04**

### CODE REVIEW

File: app/src/App.css [hunk 1]
```diff
@@ -13,9 +13,15 @@
...
-} */
+.body {
+  margin: 0;
+}
+
+.Filter {
+  text-align: center;
+  font-size: 15px;
+  font-family: Verdana, Geneva, Tahoma, sans-serif;
+}
```
Comment (zeethefedev) [inline app/src/App.css:24]: Class name should be in lower case and hyphen delimited (eg: `done-form` instead of `DoneForm`)

File: app/src/App.js [hunk 2]
```diff
@@ -51,69 +58,108 @@ function App() {
...
+    } else if (taskChange === 40) {
+      taskChangeApplied.sort((a, b) => new Date(a.due) - new Date(b.due));
+    } else if (taskChange === 50) {
+      taskChangeApplied = taskChangeApplied.filter(task => !task.done);
+    } else if (taskChange === 60) {
+      taskChangeApplied = taskChangeApplied.filter(task => task.done);
+    }
+
+    setFilterTasks(taskChangeApplied);
+  };
```
Comment (zeethefedev) [inline app/src/App.js:84]: the filter function can be moved to the utils file. This makes the code cleaner and easier to read

File: app/src/App.js [hunk 3]
```diff
@@ -51,69 +58,108 @@ function App() {
...
+              <div style={{display:'flex', flexDirection:'column'}}>
+                <div style={{fontWeight:'bold'}}>{task.name}</div>
+                <div>{task.due}</div>
+              </div>
+            </div>
+            <div>
+              <Button variant="contained" onClick={() => handleEdit(index)} style={{ marginRight: '10px' }}>Update</Button>
+              <Button variant="contained" onClick={() => handleDelete(index)}>Delete</Button>
+            </div>
+          </div>
```
Comment (zeethefedev) [inline app/src/App.js:102]: This can be moved to `TaskList.jsx` file

File: app/src/App.js [hunk 4]
```diff
@@ -51,69 +58,108 @@ function App() {
...
+          >
+            <MenuItem value={10}>Name: A-Z</MenuItem>
+            <MenuItem value={20}>Name: Z-A</MenuItem>
+            <MenuItem value={30}>Due date: High to Low</MenuItem>
+            <MenuItem value={40}>Due date: Low to High</MenuItem>
+            <MenuItem value={50}>Priority: TODO</MenuItem>
+            <MenuItem value={60}>Priority: Done</MenuItem>
+          </Select>
+        </FormControl>
+      </Box>
```
Comment (zeethefedev) [inline app/src/App.js:153]: This can be moved to `Filter.jsx` file

File: app/src/App.css (hunk 1 above)
Comment (Baozzz04) [reply app/src/App.css:24]: fixed

File: app/src/App.js (hunk 3 above)
Comment (Baozzz04) [reply app/src/App.js:102]: fixed

Comment (Baozzz04) [reply app/src/App.js:102]: fixed

File: app/src/App.js (hunk 2 above)
Comment (Baozzz04) [reply app/src/App.js:84]: fixed

File: app/src/App.js (hunk 4 above)
Comment (Baozzz04) [reply app/src/App.js:153]: fixed

**Review by zeethefedev — APPROVED** [review]

**Pull request merged**